mongodump_passthrough/functions.yml

```diff
@@ -0,0 +1,773 @@
+      params:
+        shell: bash
+        working_dir: src/mongosync
+        script: |
+          set -o errexit
+          set -o pipefail
+          set -o xtrace
+
+          ${PREPARE_SHELL}
+
+          mise exec -- go version
+          mise exec -- go env
+
+  "run mise and npm install":
+    - command: shell.exec
+      params:
+        shell: bash
+        working_dir: src/mongosync
+        script: |
+          set -o errexit
+          set -o pipefail
+          set -o xtrace
+
+          ${PREPARE_SHELL}
+
+          mise install
+          mise exec -- npm install
+
+  # Added for mongodump_passthrough
+  #
+  # This build the mongo-tools executables, then copies mongogump and mongorestore
+  # into src/mongosync/dist.
+  "build mongodump and mongorestore":
+    - command: shell.exec
+      params:
+        working_dir: src/github.com/mongodb/mongo-tools
+        script: |
+          ${PREPARE_SHELL}
+          PATH=$PATH:$HOME
+          echo $PATH 
+          MISE_INSTALL_PATH='${workdir}/.local/bin/mise' sh ${workdir}/src/mongosync/etc/mise.run.sh
+          mise settings experimental=true
+          mise activate --shims
+          PATH=$HOME/.local/share/mise/shims:$PATH
+          # We only install Go here in order to get the system to a state where we can run mage. If
+          # more dev tools are needed, those tasks should run `mise install`.
+          mise install go
+          mise use -g go@1.24.0
+          mise exec -- go run build.go build
+          # Copy mongodump and mongorestore to mongosync/dist
+          mkdir -p ${workdir}/src/mongosync/dist
+          cp -p bin/mongodump bin/mongorestore ${workdir}/src/mongosync/dist
+
+  "install podman on ubuntu":
+    - command: shell.exec
+      params:
+        shell: bash
+        working_dir: .
+        script: |
+          set -o errexit
+          set -o pipefail
+          set -o xtrace
+
+          ${PREPARE_SHELL}
+
+          sudo apt-get update
+          sudo apt-get install -y podman
+
+  f_repo_with_modules_fetch:
+    - command: git.get_project
+      params:
+        directory: src/github.com/mongodb/mongo-tools
+        revisions:
+          migration-verifier: *_migration_verifier_pin
+          mongosync: *_mongosync_pin
+
+  # Under src/mongosync/resmoke we have separate directories "suite-config"
+  # and "mongodump-suite-config". To run mongodump passthrough tests we delete
+  # mongosync-related suites directories and copy the mongodump_suite-config.
+  # This keeps the mongosync suite-config/*/loggers dirs.
+  f_enable_mongodump_suite_config: &f_enable_mongodump_suite_config
+    command: shell.exec
+    params:
+      shell: bash
+      working_dir: src/mongosync/resmoke
+      script: |
+        set -o errexit
+        set -o pipefail
+        set -o xtrace
+
+        rm -rf `find suite-config -name "suites"`
+        cd mongodump-suite-config
+        tar cvf ../suite-config/a.tar .
+        cd ../suite-config
+        tar xvf a.tar
```

Comment:
Why use `tar` like this instead of `cp -r`?

Reply:
I'm more confident that I understand the behavior of tar.

"Historic versions of the cp utility had a -r option.  This implementation supports that option, however,
  its behavior is different from historical FreeBSD behavior.  Use of this option is strongly discouraged as
  the behavior is implementation-dependent."

Reply:
We only run this stuff on Linux, which has a `cp -r` that works as you'd expect. Using `tar` seems unnecessarily baroque.

Reply:
Baroque is my era.  I'm easily confused by the way cp puts things inside a destination directory if it already exists.

Reply:
I think this should be changed. This is a very idiosyncratic thing that we don't do elsewhere, AFAIK. It's time to move onto the Classical era, if not Romantic.